File: 01/p2.py
with open("input", "r") as f:
    lines = f.readlines()
    
total = 0
position = 50

for line in lines:
    if line[0] == "L":
        prev_pos = position
        position -= int(line.strip()[1:])
        # position is always positive at the end of the loop, 
        # so, if it's negative we must have crossed 0 at least once
        if position < 0:
            total += abs(position // 100)
            # if we were already at zero before, 
            # this number is over valued by 1
            if prev_pos == 0:
                total -= 1
        # check if we're now stopped at zero
        position = position % 100
        if position == 0:
            total += 1
    elif line[0] == "R":
        position += int(line.strip()[1:])
        total += position // 100
        position = position % 100

print(total)

File: 01/p2.cpp
#include <iostream>
#include <fstream>
#include <string>
using namespace std;

int main() {
    ifstream f("input", ifstream::in);
    string line;

    int position = 50;
    int total = 0;
    int prev_pos;
    while (f >> line) {
        if (line.at(0) == 'L') {
            prev_pos = position;
            position -= stoi(line.substr(1));

            if(position < 0) {
                int floor_q = position / 100;
                if (position % 100 != 0) floor_q -= 1;
                
                total += abs(floor_q);
                if (prev_pos == 0) {
                    total -= 1;
                }
            }
            position = (position % 100 + 100) % 100; // ensure positive modulus
            if (position == 0)
                total += 1;

        } else if(line.at(0) == 'R') {
            position += stoi(line.substr(1));
            total += position / 100;
            position = position % 100;
        }
    }
    cout << total << endl;
    return 0;
}